### DISCUSSION TITLE
Naming of repository functions

### DISCUSSION THREAD

**Discussion by garma83** [CodeDredd/pinia-orm#774] [Ideas]
Hi! First off thanks for your great work :)

I find the naming of the repository functions somewhat confusing. Or maybe not the naming, but what they do. 

Could you share the reasoning behind the current way, especially with regards to the following:
- why does insert disregard relationships?
- why is it that I sometimes need to use query() and sometimes I dont? e.g. Model.query().where().get(), but also Model.where().update()
- why is it Model.save() but also Model.query().update()? Shouldn't there be a Model.update() as well? 
- why can I not use all() in a query like this: Model.query().where('id',id).all()? Either remove all() so that it's clear that that doesnt work, or make it so that it is similar to get(). (vuex-orm uses all() in queries, so this leads to a lot of bugs on my end)

What I would expect:

insert -> insert a new instance, including all relations. Throw an error if the id already exists.
update -> update an existing instance, including replacing relations. throws an error if entity does not exist
save -> insert or update depending on whether Id exists

Would it be an option to bring this more in line with the corresponding vuex-orm package, which im migrating away from?

**Comment by CodeDredd** [comment]
Thanks again 😄 .... you are finding good points. Should have done that before v1.
I fear that this change are introducing breaking changes....well i will find out when i start aligning.

The reason for having this that way, is no reaason ^^....it's still legacy from the vuex-orm-next code base.

> why does insert disregard relationships?

Dunno....well if i look at the code i know. This is legacy code.

> why is it that I sometimes need to use query() and sometimes I dont? e.g. Model.query().where().get(), but also Model.where().update()

you are right i need here more silver lining. Although your example doesn't need query(). From my point of few i dont want to have repository all query functions, because it adds only not necessary code to the bundle. since most functions in repo are just shortcuts not to use `useRepo.query()`
UPDATE: After thinking i just add the missing ones into repository...because almost all are there.

> why is it Model.save() but also Model.query().update()? Shouldn't there be a Model.update() as well?

well....yes....but right now update only works with query chain builder. So from that perspetive it makes sense not haveing it in repository. 
actually it's fine because its aligning with https://laravel.com/docs/9.x/eloquent#updates . So you can use `save()` for update.
But including thrwoing an error seems to be something to have :)

>why can I not use all() in a query like this: Model.query().where('id',id).all()? Either remove all() so that it's clear that that doesnt work, or make it so that it is similar to get(). (vuex-orm uses all() in queries, so this leads to a lot of bugs on my end)

well all() should never be the same like get()....otherwise i dont need `all()` at all. in the docs its stated clear that `all()` ignores alll query constraints. It also doesn't use the query builder. so it's also a performance difference.  Maybe it makes sense to remove `all()` from query and move it to `repository`. This is sadly a breaing change.....seems i have to start on v2 ^^
`all()` -> get store results -> hydrate them
`get()` _> get store results -> execute query builder -> hydrate them

**Reply by garma83** [reply]
thanks for the reply :) My experience was with vuex-orm and not with vuex-orm-next, I haven't actually worked with that.

I understand the concerns about a breaking change. Let me know if I can help think along somehow.

I think the 'insert' one is one of the main confusing ones, since it basically renders that one not very useful. It also seems like that one can be fixed without breaking changes? it would have to do an id check, and then just use save() maybe?

the other one would be the error messaging in insert() and update(), that also would save me some headaches :) that is also non-breaking? It could even be just a console.warn()

Happy holidays!

**Reply by CodeDredd** [reply]
Happy new year first :)
And econd i opened an PR to address this for now without breaking anything. It's more an refactoring and bringing missing functinality.
I added your console warning also. You can look at the PR #790
